Description: Fix EA header related file move/delete errors
 Check for empty EA slots and fix log messag missing parameter
Author: Frank Lahm <franklahm@googlemail.com>
Origin: Commit: cea5233c2f942bb5c90612d0c313b3c91fa7d476
Bug: http://sourceforge.net/mailarchive/message.php?msg_id=28028073
Bug-Debian: http://bugs.debian.org/648792
Last-Update: 2011-11-30

--- a/libatalk/vfs/ea.c
+++ b/libatalk/vfs/ea.c
@@ -514,13 +514,15 @@
     unsigned int count = 0;
 
     if (ea->ea_count == 0) {
-        LOG(log_error, logtype_afpd, "ea_delentry('%s'): illegal ea_count of 0 on deletion");
+        LOG(log_error, logtype_afpd, "ea_delentry('%s'): illegal ea_count of 0 on deletion",
+            attruname);
         return -1;
     }
 
     while (count < ea->ea_count) {
         /* search matching EA */
-        if (strcmp(attruname, (*ea->ea_entries)[count].ea_name) == 0) {
+        if ((*ea->ea_entries)[count].ea_name &&
+            strcmp(attruname, (*ea->ea_entries)[count].ea_name) == 0) {
             free((*ea->ea_entries)[count].ea_name);
             (*ea->ea_entries)[count].ea_name = NULL;
 
